### PR TITLE
Separate enums from models

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Follow up of #446

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
I wonder if we have some criteria to separate more of them from `models.go`? Reduction of the size of `models.go` may solve the issue of big files in which the intellisense stops working and will increase user experience a lot

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT could you please have a look?

File: test/src/tests/generated/additionalproperties/enums.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package additionalproperties
```
Comment (jhendrixMSFT) [inline test/src/tests/generated/additionalproperties/enums.go:1]: You should avoid generating empty enum files.

**Comment by jhendrixMSFT** [comment]
This is using `gopls` for intellisense?  Do you know if large files contributes to this problem?  I'm not against carving this up even more, however it would be nice to know that the work has some benefit.

Also, can you please sync with the Terraform codegen folk to make sure this split isn't going to cause any problems?

**Comment by ArcturusZhang** [comment]
Hi @jhendrixMSFT I made some changes to make it does not generate `enums.go` when there were not any

**Comment by fearthecowboy** [comment]
# 🤖 AutoRest automatic publish job 🤖
## success (version: 2.1.148)

**Pull request merged**